changelog/README.md

```diff
@@ -0,0 +1,59 @@
+# Multi-file running changelog
+
+To avoid merge conflicts in the CHANGELOG.md file we accepted the multi-file running changelog strategy.
+
+To follow this strategy you should create a `changelog/unreleased/${PR_NUMBER}.json` file for every PR like:
+
+```
+[
+  {
+    "type": "Features",
+    "title": "Description of changes"
+  }
+]
+```
+
+`type` it is a header of changes in the `CHANGELOG.md`. `title` it is a text of changes
+
+If you have implemented some features or bugfixes you should describe all of them:
+
+```
+[
+  {
+    "type": "Features",
+    "title": "Description of changes 1"
+  },
+  {
+    "type": "Bug fixes and improvements",
+    "title": "Description of changes 2"
+  }
+]
+```
+
+Every release the release train app will:
+
+* collect all files from `changelog/unreleased`
+* assemble the changelog with versions of dependencies like:
+```
+## Mapbox Navigation SDK 1.1.1 - 13 December, 2022
+### Changelog
+[Changes between v1.1.0 and v1.1.1](https://github.com/mapbox/mapbox-navigation-android/compare/v1.1.0...v1.1.1)
+
+#### Features
+- Feature 1 [#1234](https://github.com/mapbox/mapbox-navigation-android/pull/1234)
+- Feature 2 [#2345](https://github.com/mapbox/mapbox-navigation-android/pull/2345)
+
+#### Bug fixes and improvements
+- Bugfix 3 [#3456](https://github.com/mapbox/mapbox-navigation-android/pull/3456)
+- Bugfix 4 [#4567](https://github.com/mapbox/mapbox-navigation-android/pull/4567)
+
+### Mapbox dependencies
+This release depends on, and has been tested with, the following Mapbox dependencies:
+- Mapbox Maps SDK `v10.8.0` ([release notes](https://github.com/mapbox/mapbox-maps-android/releases/tag/v10.8.0))
+- Mapbox Navigation Native `v115.0.1`
+- Mapbox Core Common `v23.0.0`
+- Mapbox Java `v6.8.0` ([release notes](https://github.com/mapbox/mapbox-java/releases/tag/v6.8.0))
+- Mapbox Android Core `v5.0.2` ([release notes](https://github.com/mapbox/mapbox-events-android/releases/tag/core-5.0.2))
+```
+* write the changelog to the `CHANGELOG.md` file
```

Comment:
capturing @tomaszrybakiewicz comment.

Instead of generating CHANGELOG during release, do this on PR merge.

I'm also +1'ing that. It is nice to see a running changelog, instead of needing to dig through each new file created. 

Considering that may re-introduce the merge conflict issue, maybe it could be part of a script where we can say `sh scripts/show-unreleased-changelog.sh`?

Reply:
an idea is to have a file that represent the latest. every time a new pull request is merged the other one is deleted. 

```
pr 01 is merged
01-changelog.md

pr 02 is merged
01-changelog.md is deleted
02-changelog.md is created
```

not the best solution considering that the unreleased changelog link would have a short life https://github.com/mapbox/mapbox-navigation-android/blob/main/unreleased/01-changelog.md

Reply:
Do we want to keep current not released version of running changelog publicly available? 
If not, maybe we can generate current changelog by request inside the release train app?

Reply:
>After merging a PR,

it means that changelog must be merged separately, in another commit, out of PR merge. It seems risky: make it in 1 step (aka transaction) safer, than drop to a few

Reply:
also, this way we will have 2x commits in the branch (1 is changed + 1 changelog)

Reply:
Do we really need unreleased changelog in one file before release?

Reply:
> it means that changelog must be merged separately, in another commit, out of PR merge. It seems risky: make it in 1 step (aka transaction) safer, than drop to a few

@RingerJK It's as risky as Release Train or CI release workflows making additional commits.

> also, this way we will have 2x commits in the branch (1 is changed + 1 changelog)

And why is that a problem?

Reply:
I think we can add one more job for the main branch to generate a running changelog file. This job will be executed after every merge